File: core/src/main/java/co/infinum/collar/annotations/ConvertAttribute.java
package co.infinum.collar.annotations;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Should be used with {@link ConvertAttributes}
 */
@Retention(RetentionPolicy.RUNTIME)
public @interface ConvertAttribute {

    String value();

    String defaultValue() default "";

    /**
     * Keep this event in entire app scope
     */
    boolean isSuper() default false;
}

File: annotations/src/main/kotlin/co/infinum/collar/annotations/ConvertAttribute.kt
package co.infinum.collar.annotations

/**
 * Should be used with [ConvertAttributes]
 */
@Retention(AnnotationRetention.RUNTIME)
annotation class ConvertAttribute(
    val value: String, val defaultValue: String = "",
    /**
     * Keep this event in entire app scope
     */
    val isSuper: Boolean = false
)
